actions/get-usns/entrypoint/main.go

```diff
@@ -0,0 +1,339 @@
+package main
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"html"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"net/url"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/mmcdole/gofeed"
+)
+
+var distroToVersionRegex map[string]string = map[string]string{
+	"jammy":  `22\.04`,
+	"focal":  `20\.04`,
+	"bionic": `18\.04`,
+}
+
+type USN struct {
+	AffectedPackages []string `json:"affected_packages"`
+	CVEs             []CVE    `json:"cves"`
+	Title            string   `json:"title"`
+	ID               string   `json:"id"`
+	URL              url.URL  `json:"-"`
+	URLString        string   `json:"url"`
+}
+
+type CVE struct {
+	Description string `json:"description"`
+	Title       string `json:"title"`
+	URL         string `json:"url"`
+}
+
+func main() {
+	var config struct {
+		Distro       string
+		LastUSNsJSON string
+		Output       string
+		PackagesJSON string
+		RSSURL       string
+	}
+
+	flag.StringVar(&config.LastUSNsJSON,
+		"last-usns",
+		"",
+		"JSON array of last known USNs")
+	flag.StringVar(&config.RSSURL,
+		"feed-url",
+		"https://ubuntu.com/security/notices/rss.xml",
+		"URL of RSS feed")
+	flag.StringVar(&config.PackagesJSON,
+		"packages",
+		"",
+		"JSON array of relevant packages")
+	flag.StringVar(&config.Distro,
```

Comment:
I wonder if it would be possible to get rid of the `patched-usns.json` file if we input the packages and their versions to this action. In that case, you could grab the USNs from some time period, maybe the last month, and then just compare affected packages and their versions. If you haven't upgraded to at least the version specified in the USN, then consider it an unpatched USN.

Reply:
Filed https://github.com/paketo-buildpacks/jammy-tiny-stack/issues/7 to investigate this.